### ISSUE TITLE
x/tools/gopls: reconsider var hover behavior with constant initializer

### ISSUE THREAD

**Issue by findleyr** [golang/go#58220]
In #45802, we added hover content for variables whose initializer is an int literal (e.g. `var foo = 123`).

As I rewrite the hover logic, I think this is a misfeature: it is misleading to make it look like the variable has a known value.

Instead, we should support hovering over hex/octal/binary literals.

**Comment by bcmills** [comment]
Ideally we would display the variable's value if (and only if) no other writes occur in the workspace. (But that's a much harder analysis.)

**Comment by adonovan** [comment]
I agree it's harder. For unexported vars it only requires looking at a single type-checked package but that's still a lot more work than hover needs to do in the common case, after Rob's recent work on it.
